docs/src/how-to/programmers-guide.rst

```diff
@@ -0,0 +1,202 @@
+
+  Running the preceding command generates an executable ``TensileTests``, which can be further used to run the tests.
+
+- Manually: To build and run the tests manually, see the commands in `tasks.py <https://github.com/ROCm/Tensile/blob/develop/tasks.py>`_.
+  For advanced usage like filtering or repeating test cases, see the `gtest documentation <https://github.com/google/googletest/blob/main/docs/advanced.md>`_.
+
+===============
+Static analysis
+===============
+
+------
+Python
+------
+
+To run all static analysis, use the top-level ``tox`` label ``static``:
+
+.. code-block::
+
+   tox run -m static
+
+.. note::
+   The preceding command might reformat your code, so make sure to commit your changes after running the command.
+
+**Linting** is evaluated using `flake8 <https://flake8.pycqa.org/en/latest/>`_ and **formatting** is conducted using `black <https://black.readthedocs.io/en/stable/>`_ and `isort <https://pycqa.github.io/isort/>`_. To run a check in isolation, either refer to `tox.ini <https://github.com/ROCm/Tensile/blob/develop/tox.ini>`_ or use one the following commands:
+
+.. code-block::
+
+   tox run -e lint
+   tox run -e format     # add `-- --check` to check formatting without applying changes
+   tox run -e isort      # add `-- --check` to check imports without applying changes
+
+
+.. tip::
+
+   To ensure consistent formatting, we recommend you to set up the editor to **format on save** using the same formatter settings as in `tox.ini <https://github.com/ROCm/Tensile/blob/develop/tox.ini>`_. Either way, ensuring to commit changes after running static analysis reduces wait times caused by simple CI failures.
+
+---
+C++
+---
+
+**Formatting** is conducted using `clang-format <https://clang.llvm.org/docs/ClangFormatStyleOptions.html>`_.
+The following command formats all given files, however, we recommend you to setup the editor to *format on save*.
+
+.. code-block::
+
+   clang-format -i style=file <files>
+
+Styling rules are configured in `.clang-format <https://github.com/ROCm/Tensile/blob/develop/.clang-format>`_.
+
+=========
+Profiling
+=========
+
+------
+Python
+------
+
+To enable profiling, use the ``@profile`` decorator, which must be imported from the ``Tensile.Utilities.Profile`` module. Under the hood, the decorator wraps the function in a `cProfile <https://docs.python.org/3/library/profile.html#module-cProfile>`_ context and generates a ``.prof`` file inside the ``profiling-results-<date>`` directory.
+
+.. note::
+   Nested profiling is NOT supported due to the existing limitation with the profiling decorator. This implies that if `func1` calls `func2` in a loop, and both are marked for profiling, the resulting ``.prof`` file for `func1` will display incorrect results.
+
+========================
+Building documentation
+========================
+
+Tensile uses https://github.com/ROCm/rocm-docs-core as the documentation engine, which itself wraps Read the Docs and Sphinx.
```

Comment:
@bstefanuk Can we replace this para with just the link https://rocm.docs.amd.com/en/latest/contribute/contributing.html?
I am not sure if we need to talk about rocm-docs-core here as users don't have to do anything there.

Reply:
Done.